cuda_core/cuda/core/experimental/_module.py

```diff
@@ -215,6 +220,36 @@ def attributes(self) -> KernelAttributes:
             self._attributes = KernelAttributes._init(self._handle)
         return self._attributes
 
+    def _get_arguments_info(self, param_info=False) -> tuple[int, list[ParamInfo]]:
+        attr_impl = self.attributes
+        if attr_impl._backend_version != "new":
+            raise NotImplementedError("New backend is required")
+        arg_pos = 0
+        param_info_data = []
+        while True:
```

Comment:
Maybe (sorry I overlooked this before):

```python
        for arg_pos in itertools.count():
```

Then you don't need `arg_pos = 0` above and `arg_pos = arg_pos + 1` below.